### ISSUE TITLE
InvRef's add_item and remove_item should also return index

### ISSUE THREAD

**Issue by kromka-chleba** [minetest/minetest#15608]
### Problem

Currently when you add or remove an item from an intenvory using 'inv:add_item' and 'inv:remove_item' only the leftover stack is returned. It may be useful to also return the index to/from which the itemstack was added.

### Solutions

Returning the index?

### Alternatives

Rewriting the functionality of `add_item` and `remove_item` using the current API but also returning the index.

**Comment by cx384** [comment]
It would be good if you provide some concrete use cases to make it clearer how useful this feature is.

Also, if am not wrong, `add_item` may split the stack and add it to the inventory at multiple indexes.
So it's not that straight forward.

**Comment by appgurueu** [comment]
> Also, if am not wrong, `add_item` may split the stack and add it to the inventory at multiple indexes. So it's not that straight forward.

That's correct, see `InventoryList::addItem` and `InventoryList::removeItem` respectively; `remove_item` may take from multiple slots as well.

I think at this point it's not worth returning a list of modified indices or similar and modders should prefer to write this simple something like 5 to 10 line piece of code out themselves so they can get precisely what they want.

**Comment by kromka-chleba** [comment]
>It would be good if you provide some concrete use cases to make it clearer how useful this feature is.

I'm writing a shelf mod where item entities need to be synchronized with inventory of the shelf node. Inventory slot positions correspond to certain places on the shelf and knowing the exact inventory slot makes it easier to manage.

```
Also, if am not wrong, add_item may split the stack and add it to the inventory at multiple indexes.
So it's not that straight forward.
```

Weird, I did not observe this behavior. Would be good if this was documented, because all the API says is `add_item(listname, stack): add item somewhere in list, returns leftover ItemStack.` if I knew that I wouldn't use the function in the first place.

>I think at this point it's not worth returning a list of modified indices or similar and modders should prefer to write this simple something like 5 to 10 line piece of code out themselves so they can get precisely what they want.

Well then, that's what I'll do. Just would be good if the documentation said precisely what the function do because the current description is misleading.

**Comment by appgurueu** [comment]
> Just would be good if the documentation said precisely what the function do because the current description is misleading.

I agree, I had to dig into the sources to be sure how of their behavior because the docs weren't helpful (though I had a hunch).
